src/mono/mono/mini/mini-arm64.c

```diff
@@ -3920,8 +3920,18 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 		// case OP_XCONCAT:
 		// 	arm_neon_ext_16b(code, dreg, sreg1, sreg2, 8);
 		// 	break;
-		
-			/* BRANCH */
+		case OP_ARM64_USHL: {
+			arm_neon_ushl (code, get_vector_size_macro (ins), get_type_size_macro (ins->inst_c1), dreg, sreg1, sreg2);
+			break;
+		}
+		case OP_ARM64_EXT_IMM: {
```

Comment:
I looked into how `OP_XCONCAT` is implemented for llvm backend and it doesn't seem to be doing the same thing as `EXT (vector)` instruction: https://github.com/dotnet/runtime/blob/ed09ae533153396321a7a4c0c8697f0e1a6f5552/src/mono/mono/mini/mini-llvm.c#L8171 and further following the implementation of `concatenate_vectors` it is doing a simple concatenation of two vectors, while in case of `EXT` we actually need a shuffle which produces same-sized result. 
For this reason I left the code unchanged.